packages/core-data/src/batch-support.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { memoize } from 'lodash/function';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { dispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import './batch-processing';
+import { STATE_ERROR } from './batch-processing/constants';
+
+async function batchProcessor( requests, batch ) {
+	if ( batch.state === STATE_ERROR ) {
+		throw {
+			code: 'transaction_failed',
+			data: { status: 500 },
+			message: 'Transaction failed.',
+		};
+	}
+
+	const response = await apiFetch( {
+		path: '/v1/batch',
+		method: 'POST',
+		data: {
+			validation: 'require-all-validate',
+			requests: requests.map( ( options ) => ( {
+				path: options.path,
+				body: options.data,
+				method: options.method,
+				headers: options.headers,
+			} ) ),
+		},
+	} );
+
+	const failed =
+		response.failed ||
+		response.responses.filter(
+			( { status } ) => status < 200 || status >= 300
+		).length;
+	if ( failed ) {
+		throw response.responses.map( ( itemResponse ) => {
+			// The REST API returns null if the request did not have an error.
+			return itemResponse === null
+				? {
+						code: 'transaction_failed',
+						data: { status: 400 },
+						message: __(
+							'This item could not be saved because another item encountered an error when trying to save.'
+						),
+				  }
+				: itemResponse.body;
+		} );
+	}
+
+	return response.responses.map( ( { body } ) => body );
+}
+
+const determineBatchSize = memoize( async () => {
+	const response = await apiFetch( {
+		path: '/v1/batch',
+		method: 'OPTIONS',
+	} );
+	return response.endpoints[ 0 ].args.requests.maxItems;
+} );
+
+// Ensure batch-processing store is available for dispatching
+setTimeout( () => {
+	dispatch( 'core/__experimental-batch-processing' ).registerProcessor(
```

Comment:
This is global and applies to all registries, it shouldn't be the case. I think the issues comes from the "registerProcessor" API. Why do we need to register a processor at all and why are these tracked in the store.

Reply:
Sorry, picking this work up from @adamziel and not sure what you mean here @youknowriad! What you would replace registering a processor with?

Reply:
@adamziel wrote some docs on `@wordpress/batch-processing` here:

https://github.com/WordPress/gutenberg/pull/27241/files#diff-15ed4b1ffcab0ae3c62807b5d98ed60ea36bfacd841f965cdac183d69d37ce66

My understanding is that the processor concept makes it so that `@wordpress/batch-processing` is a generic way to batch async work and not something that's reliant on `/v1/batch`.

I'm not against removing this abstraction and instead calling `apiFetch( '/v1/batch' )` directly from `saveEntityRecords`. In general I think it's better to hold off on adding abstractions until we're sure that we need it. Is this what you mean @youknowriad?

Reply:
Hey @noisysocks my comment was not clear it's not the abstraction that bothers me but more the shape of the API (relying on a store and strings).

For example, an API like that would be more appealing to me:

```
// Create the "processor"
const processor = createProcessor( inputs => doSomethingWithInputs );

// Create the "batch" to be kept  in core-data if it's something that need to persist between action calls
// or just use  a local variable if the batching happens within the same function like desired here `__experimentalBatchSaveEntityRecords`
const batch = createBatch( processor );

batch.add( input )
batch.add( input2 )

const result = batch.flush()
```

Reply:
Mmm! Very good. I'll have a play.